### ISSUE TITLE
Daemon stops respond after umount error

### ISSUE THREAD

**Issue by ybznek** [OpenSCAP/openscap-daemon#34]
Sorry, I still don't have enough time to provide more useful information.

While testing, it often(but not always) stops at

```
# atomic scan --all
```
It stops responds and it is not killable by ctrl+c (its very annoying!). When I close terminal and try to repeat command ```# atomic scan --all```, I will get
```
ERROR:dbus.proxies:Introspect error on :1.1837:/OpenSCAP/daemon: dbus.exceptions.DBusException: org.freedesktop.DBus.Error.NoReply: Did not receive a reply. Possible causes include: the remote application did not send a reply, the message bus security policy blocked the reply, the reply timeout expired, or the network connection was broken.
```

```
journalctl -u oscapd 
```

```
Nov 26 10:31:29 zmoravec-thinkpad oscapd[16330]: DEBUG:Scanning chroot 95338760b9fb132fe29d6b9369a2333ea48ec8b71ef0e688d078a765aeac60a6
Nov 26 10:31:29 zmoravec-thinkpad oscapd[16330]: DEBUG:Starting evaluation with command 'oscap oval eval --report /var/tmp/image-scanner/reports/95338760b9fb132fe29d6b9369a2333ea48ec8b71ef0e688d078a765aeac60a6.html --results /var/tmp/image-scanner/reports/95338760b9fb132fe29d6b9369a2333ea48ec8b71ef0e688d078a765aeac60a6.xml /var/tmp/image-scanner/com.redhat.rhsa-RHEL7.xml.bz2'.
Nov 26 10:31:31 zmoravec-thinkpad oscapd[16330]: DEBUG:"POST /run/docker.sock/v1.17/commit?container=844af2ebf856b595fee7de44f06276f08ed857eac8d880ffc7eeaeb6f476a7c0 HTTP/1.1" 201 74
Nov 26 10:31:31 zmoravec-thinkpad oscapd[16330]: DEBUG:Scanned chroot for image 7ad37622e7b6a1747b3b61448bf747f05d3d84d1d7b04f83aa58d3c8d5b00829 completed in 1.77401304245 seconds
Nov 26 10:31:31 zmoravec-thinkpad oscapd[16330]: DEBUG:Scanned chroot for image 95338760b9fb132fe29d6b9369a2333ea48ec8b71ef0e688d078a765aeac60a6 completed in 1.89656019211 seconds
Nov 26 10:31:31 zmoravec-thinkpad oscapd[16330]: DEBUG:"POST /run/docker.sock/v1.17/containers/create HTTP/1.1" 201 90
Nov 26 10:31:31 zmoravec-thinkpad oscapd[16330]: DEBUG:"GET /run/docker.sock/v1.17/containers/77372c1ed2f51ed2274d5306b69b75c775c2bff6dcb813d6f0b55173c99573f3/json HTTP/1.1" 200 None
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: DEBUG:Created scanning chroot in 6.61336708069 seconds
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: INFO:844af2ebf856b595fee7de44f06276f08ed857eac8d880ffc7eeaeb6f476a7c0 is Fedora release 22 (Twenty Two)
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: Exception in thread 844af2ebf856b595fee7de44f06276f08ed857eac8d880ffc7eeaeb6f476a7c0:
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: Traceback (most recent call last):
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: File "/usr/lib64/python2.7/threading.py", line 813, in __bootstrap_inner
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: self.run()
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: File "/usr/lib64/python2.7/threading.py", line 766, in run
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: self.__target(*self.__args, **self.__kwargs)
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: File "/usr/lib/python2.7/site-packages/openscap_daemon/cve_scanner/cve_scanner.py", line 301, in search_containers
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: f.DM.unmount_path(f.dest)
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: File "build/bdist.linux-x86_64/egg/Atomic/mount.py", line 162, in unmount_path
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: raise ValueError(r.stderr)
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: ValueError: umount: /tmp/77372c1ed2f51ed2274d: target is busy
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: (In some cases useful info about processes that
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: use the device is found by lsof(8) or fuser(1).)
Nov 26 10:31:32 zmoravec-thinkpad oscapd[16330]: DEBUG:"DELETE /run/docker.sock/v1.17/containers/037537c89f5b93c2a12e?link=False&force=False&v=False HTTP/1.1" 204 0
Nov 26 10:31:33 zmoravec-thinkpad oscapd[16330]: DEBUG:"DELETE /run/docker.sock/v1.17/containers/36f8514e03f5a86d8f5b?link=False&force=False&v=False HTTP/1.1" 204 0
```

When it freezes, I can see usually something like that in ```mount```
```
/dev/mapper/docker-253:2-2622353-2a0168f5e564072a8a40e59c402cefa872bcddc465c050069b4a484ae6403aaf on /tmp/2a0168f5e564072a8a40 type ext4 (ro,nosuid,nodev,relatime,context="system_u:object_r:svirt_sandbox_file_t:s0:c287,c651",stripe=16,data=ordered)
```
```

**Comment by ybznek** [comment]
I think, daemon should not stop respond after some error. We can also try unmount after some time. I don't know what use the device, I can add some debug info later to determine problem.

**Comment by ybznek** [comment]
Some new **oscapd** log, 0f7af34baa23 ~ SPC
```
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: DEBUG:"DELETE /run/docker.sock/v1.17/images/566dfd048e949a1e84c82affb0b3136f18e6f2e8e41ee59f26e313121f977b25?noprune=False&force=False HTTP/1.1" 409 101
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: Exception in thread 0f7af34baa23037898ae750b6e53bb47c938b55174e3b8ed4ea54ff08579c29f:
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: Traceback (most recent call last):
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib64/python2.7/threading.py", line 813, in __bootstrap_inner
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: self.run()
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib64/python2.7/threading.py", line 766, in run
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: self.__target(*self.__args, **self.__kwargs)
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib/python2.7/site-packages/openscap_daemon/cve_scanner/cve_scanner.py", line 302, in search_containers
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: f.DM._clean_temp_container_by_path(f.dest)
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib/python2.7/site-packages/Atomic/mount.py", line 537, in _clean_temp_container_by_path
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: self._clean_tmp_image()
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib/python2.7/site-packages/Atomic/mount.py", line 461, in _clean_tmp_image
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: self.client.remove_image(self.tmp_image)
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib/python2.7/site-packages/docker/client.py", line 826, in remove_image
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: self._raise_for_status(res)
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: File "/usr/lib/python2.7/site-packages/docker/client.py", line 119, in _raise_for_status
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: raise errors.APIError(e, response, explanation=explanation)
Dec 01 12:27:46 zmoravec-thinkpad oscapd[569]: APIError: 409 Client Error: Conflict ("Conflict, cannot delete b10032cad63b because the container 0f7af34baa23 is using it, use -f to force")
```

**Comment by mpreisler** [comment]
This was fixed by #42
